src/map/lua/lua_baseentity.cpp

```diff
@@ -1811,6 +1811,240 @@ inline int32 CLuaBaseEntity::pathTo(lua_State* L)
+*  Notes   : keep calling this function to keep the position
+************************************************************************/
+inline int32 CLuaBaseEntity::pathInfrontTarget(lua_State* L)
+{
+    TPZ_DEBUG_BREAK_IF(m_PBaseEntity == nullptr);
+    TPZ_DEBUG_BREAK_IF(m_PBaseEntity->objtype == TYPE_PC);
+    TPZ_DEBUG_BREAK_IF(lua_isnil(L, 1) || !lua_isuserdata(L, 1));
+
+    if (m_PBaseEntity != nullptr)
+    {
+        CLuaBaseEntity* PLuaBaseEntity = Lunar<CLuaBaseEntity>::check(L, 1);
+        CBattleEntity* PEntity = (CBattleEntity*)PLuaBaseEntity->GetBaseEntity();
+
+        if (PEntity != nullptr)
+        {
+            if (m_PBaseEntity->PAI->PathFind)
+            {
+                m_PBaseEntity->PAI->PathFind->PathInfrontTarget(PEntity->loc.p, PEntity->m_ModelSize);
+            }
+        }
+    }
+    
+    return 0;
+}
+
+/************************************************************************
+*  Function: pathToSafeDistance()
+*  Purpose : Makes a non-PC move to a safe distance away from a target
+*  Example : mob:pathToSafeDistance(mob:getTarget())
+*  Notes   : keep calling this function to keep the position
+************************************************************************/
+inline int32 CLuaBaseEntity::pathToSafeDistance(lua_State* L)
+{
+    TPZ_DEBUG_BREAK_IF(m_PBaseEntity == nullptr);
+    TPZ_DEBUG_BREAK_IF(m_PBaseEntity->objtype == TYPE_PC);
+    TPZ_DEBUG_BREAK_IF(lua_isnil(L, 1) || !lua_isuserdata(L, 1));
+
+    if (m_PBaseEntity != nullptr)
+    {
+        CLuaBaseEntity* PLuaBaseEntity = Lunar<CLuaBaseEntity>::check(L, 1);
+        CBattleEntity* PEntity = (CBattleEntity*)PLuaBaseEntity->GetBaseEntity();
+
+        if (PEntity != nullptr)
+        {
+            if (m_PBaseEntity->PAI->PathFind)
+            {
+                m_PBaseEntity->PAI->PathFind->PathToSafeDistance(PEntity->loc.p, PEntity->m_ModelSize);
+            }
+        }
+    }
+    
+    return 0;
+}
+
+/************************************************************************
+*  Function: pathToMeeleRange()
+*  Purpose : Makes a non-PC move or keep within meele range of a target
+*  Example : mob:pathToMeeleRange(mob:getTarget())
+*  Notes   : keep calling this function to keep the position
+************************************************************************/
+inline int32 CLuaBaseEntity::pathToMeleeRange(lua_State* L)
+{
+    TPZ_DEBUG_BREAK_IF(m_PBaseEntity == nullptr);
+    TPZ_DEBUG_BREAK_IF(m_PBaseEntity->objtype == TYPE_PC);
+    TPZ_DEBUG_BREAK_IF(lua_isnil(L, 1) || !lua_isuserdata(L, 1));
+
+    if (m_PBaseEntity != nullptr)
+    {
+        CLuaBaseEntity* PLuaBaseEntity = Lunar<CLuaBaseEntity>::check(L, 1);
+        CBattleEntity* PEntity = (CBattleEntity*)PLuaBaseEntity->GetBaseEntity();
+
+        if (PEntity != nullptr)
+        {
+            if (m_PBaseEntity->PAI->PathFind)
+            {
+                m_PBaseEntity->PAI->PathFind->PathToMeeleRange(PEntity->loc.p, PEntity->m_ModelSize);
```

Comment:
Meele leftover

Reply:
Oh damn it, cheers

Reply:
ty Xaver, i would have mised that for ages, as im busy with other stuff, should be good now, ill keep an eye on it this time.